File: src/main/java/com/sparkTutorial/rdd/sumOfNumbers/SumOfNumbersSolution.java
package com.sparkTutorial.rdd.sumOfNumbers;

import org.apache.log4j.Level;
import org.apache.log4j.Logger;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

import java.util.Arrays;

public class SumOfNumbersSolution {

    public static void main(String[] args) throws Exception {
        Logger.getLogger("org").setLevel(Level.OFF);

        SparkConf conf = new SparkConf().setAppName("primeNumbers").setMaster("local[*]");
        JavaSparkContext sc = new JavaSparkContext(conf);

        JavaRDD<String> lines = sc.textFile("in/prime_nums.text");
        JavaRDD<String> numbers = lines.flatMap(line -> Arrays.asList(line.split("\\s+")).iterator());

        JavaRDD<String> validNumbers = numbers.filter(number -> !number.isEmpty());

        JavaRDD<Integer> intNumbers = validNumbers.map(number -> Integer.valueOf(number));

        System.out.println("Sum is: " + intNumbers.reduce((x, y) -> x + y));
    }
}

File: src/main/scala/com/sparkTutorial/rdd/sumOfNumbers/SumOfNumbersSolution.scala
package com.sparkTutorial.rdd.sumOfNumbers

import org.apache.log4j.Level
import org.apache.log4j.Logger
import org.apache.spark.{SparkConf, SparkContext}

object SumOfNumbersSolution {

    def main(args: Array[String]) {
        Logger.getLogger("org").setLevel(Level.OFF)

        val conf = new SparkConf().setAppName("primeNumbers").setMaster("local[*]")
        val sc = new SparkContext(conf)

        val lines = sc.textFile("in/prime_nums.text")

        val numbers = lines.flatMap(line => line.split("\\s+"))

        val validNumbers = numbers.filter(number => !number.isEmpty)

        val intNumbers = validNumbers.map(number => number.toInt)

        println("Sum is: " + intNumbers.reduce((x, y) => x + y))
    }
}
